### PR TITLE
DBI Hotfix

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
Restore `eval_dbr_unitary`
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibo/models/variational.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def minimize(
             the ``OptimizeResult``, for ``'cma'`` the ``CMAEvolutionStrategy.result``,
             and for ``'sgd'`` the options used during the optimization.
         """
+        print("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")
```
Comment (andrea-pasquale) [inline src/qibo/models/variational.py:85]:
```suggestion
```

File: src/qibo/models/variational.py [hunk 2]
```diff
@@ -96,7 +97,7 @@ def minimize(
             loss = lambda p, c, h: dtype(loss_func(p, c, h))
         elif method != "sgd":
             loss = lambda p, c, h: self.hamiltonian.backend.to_numpy(loss_func(p, c, h))
-
+        print("JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ", options)
```
Comment (andrea-pasquale) [inline src/qibo/models/variational.py:100]:
```suggestion
```

File: src/qibo/models/dbi/double_bracket.py [hunk 3]
```diff
@@ -100,36 +100,67 @@ def __init__(
...
     ):
-        """Performs one double bracket rotation."""
+        r"""We use convention that $H' = U^\dagger H U$ where $U=e^{-sW}$ with $W=[D,H]$
+        (or depending on `mode` an approximation, see `eval_dbr_unitary`).
+        If $s>0$ then for $D = \Delta(H)$ the GWW DBR will give a $\sigma$-decrease,
+        see https://arxiv.org/abs/2206.11772."""
+
+        operator = self.eval_dbr_unitary(step, mode, d)
+        operator_dagger = self.backend.cast(
+            np.matrix(self.backend.to_numpy(operator)).getH()
```
Comment (andrea-pasquale) [inline src/qibo/models/dbi/double_bracket.py:110]:
```suggestion
            np.array(np.matrix(self.backend.to_numpy(operator)).getH())
```
I fixed this in #1269 it was creating some issues with `energy_fluctuations`.

**Review by andrea-pasquale — APPROVED** [review]
Thanks @Edoardo-Pedicillo, I confirm that now boostvqe is working as expected.

**Review by MatteoRobbiati — APPROVED** [review]
Thanks.

**Pull request merged**